[김휘민] Mission 완료 및 리팩토링 진행중

hwibaski opened the pull request:
# Mission 진행 상황 
> 완료한 미션에 대해 체크해주세요.

- [x] Mission 0A 완료
- [x] Mission 0B 완료
- [x] Mission 0C 완료
- [x] Mission 1 완료
- [x] Mission 2 완료
- [x] Mission 3 완료
- [x] Mission 4 완료
- [x] Mission 5 완료
- [x] Mission 6 완료
- [x] Mission 7 완료
- [x] Mission 8 완료

> 마크다운에서 체크를 하기 위해서는 [x] 이렇게 수정하시면 됩니다. [x ], [ x ] 가 아니라 정확하게 [x]여야 합니다. 혹은 마우스 클릭으로 체크해주세요.

# Self-Refactoring 체크리스트
> [Pre-Course 리팩토링 체크리스트](https://www.notion.so/wecode/Pre-Course-Refactoring-869f50bf6a934db98573229c600043ee)를 참고하여 리팩토링 진행해주세요.

> [Foundation Fontend 리팩토링 체크리스트](https://www.notion.so/wecode/Foundation-Frontend-Refactoring-85e629d7a22841b2bd4917c9c72a1f58)를 참고하여 리팩토링을 진행한 항목에 대해 체크를 해주세요.

- [x] 0. Pre-Course 리팩토링 체크리스트 리팩토링 완료
- [x] 1. reset.scss & common.scss
- [x] 2. sass nesting 완료
- [x] 3. className 동적 활용
- [x] 4. destructuring
- [x] 5. Boolean 데이터 타입 활용
- [x] 6. 반복되는 코드 Array.map() 활용
- [x] 7. 계산된 속성명 활용
- [x] 8. \<a\>와 \<Link\> 구분하여 사용
- [x] 9. import 순서

# Peer Reviews
- [x] Peer Reviews 받기 완료

hwibaski committed:
Add: 초기 파일 추가
- image 파일 미추가
- 공통 component 비활성화(TopNav, CoffeeCard)

hwibaski committed:
Modify: 컴포넌트 나누기, Mission 5 완료
- 각 페이지 컴포넌트화
- detail 페이지 하트 아이콘 토글화
- 댓글 삽입 기능 추가
- 목데이터로 detail 페이지 렌더링 구현

hwibaski committed:
ADD: MISSION 6 완료
- 리스트 페이지 커피별 좋아요 기능 구현

hwibaski committed:
ADD: MISSION 7 완료
- 댓글 삭제 기능 구현

hwibaski committed:
ADD: MISSION 8 완료
- detail 페이지 댓글 좋아요 기능 구현 완료

hwibaski committed:
Refactor: 스스로 리팩토링, 중복 코드 제거

hwibaski committed:
Add: Fetching Local JSON for Products

hwibaski committed:
refactor: apply pre-course refactoring

hwibaski committed:
refactor: apply front-end refactoring

wonkooklee commented on src/pages/hwiminKim/Detail/Detail.js:
```diff
@@ -1,9 +1,40 @@
 import './Detail.scss';
 
 class Detail extends Component {
+  state = {
```
🧐

hwibaski replied:
리뷰내용 : 최상위 페이지에 꼭 필요한 state 인가요?

JHyeon0 replied:
JavaScript class 문법상 이렇게 해도 상관없습니다!

wonkooklee commented on src/pages/hwiminKim/Detail/DetailCoffeeFeature.js:
```diff
@@ -0,0 +1,19 @@
+    return (
+      <section className='infoContainer'>
+        <DetailFeatureHeader description={this.props?.description} />
+        <DetailNutritionInfo nutrition={this.props?.nutrition} />
```
🤔

k-cool replied:
☔️🚑😡 할당 해주쎄어ㅎㅎㅎ 😉

wonkooklee commented on src/pages/hwiminKim/Detail/DetailFeatureHeader.js:
```diff
@@ -0,0 +1,49 @@
+
+class DetailFeatureHeader extends Component {
+  state = {
+    isToggleOn: false,
```
🤔

wonkooklee commented on src/pages/hwiminKim/Detail/DetailFeatureHeader.js:
```diff
@@ -0,0 +1,49 @@
+            <h2 className='infoTitle'>{name}</h2>
+            <p>{engName}</p>
+          </div>
+          <FontAwesomeIcon
```
🤔

wonkooklee commented on src/pages/hwiminKim/Detail/DetailNutritionInfo.js:
```diff
@@ -0,0 +1,64 @@
+
+class DetailNutritionInfo extends Component {
+  render() {
+    const servingSize = this.props.nutrition?.servingSize;
```
🤔

k-cool replied:
😨

wonkooklee commented on src/pages/hwiminKim/Detail/DetailNutritionInfo.js:
```diff
@@ -0,0 +1,64 @@
+            <p>{allergen}</p>
+          </div>
+        ) : (
+          <div className='emptyAllergyCause'></div>
```
🤔

wonkooklee commented on src/pages/hwiminKim/List/List.js:
```diff
@@ -1,9 +1,37 @@
+
   render() {
-    return <div></div>;
+    const { coldBrews, breweds } = this.state;
```
🤔

wonkooklee commented on src/pages/hwiminKim/component/ReviewList.js:
```diff
@@ -0,0 +1,73 @@
+  inputRef = React.createRef();
+  id = 0;
+  state = {
+    newReview: '',
```
🤔

k-cool commented on src/pages/hwiminKim/Detail/Detail.js:
```diff
@@ -1,9 +1,40 @@
+      <>
+        <section className='Detail'>
+          <TopsNav />
+          <DetailSubHeader category={product?.category} />
```
크으.... 이거 다시봐도 신기하네요 ㅋㅋㅋ

k-cool commented on src/pages/hwiminKim/Detail/DetailCoffeeInfo.js:
```diff
@@ -0,0 +1,20 @@
+        <DetailCoffeeFeature
+          description={this.props?.description}
+          nutrition={this.props?.nutrition}
+        />
```
this.props가 오버플로우하면 비구조화 할당 !!! 적용해보시면 신세계에여 ㅎㅎㅎ

k-cool commented on src/pages/hwiminKim/Detail/DetailFeatureHeader.js:
```diff
@@ -0,0 +1,49 @@
+  };
+
+  render() {
+    const name = this.props.description?.name;
```
여기도 비구조화 할당 거리가 보이는군요 후후 ~~;)

k-cool commented on src/pages/hwiminKim/Detail/DetailSubHeader.js:
```diff
@@ -0,0 +1,62 @@
+            <Link>홈</Link>
+          </li>
+          <li>
+            <FontAwesomeIcon
```
앜ㅋㅋ 아이콘 쓰셨다는게 이거군요 ㅋㅋㅋㅋㅋㅋ 응용력 무엇ㅋㅋㅋㅋ 
하지만 오늘 원국님이 알려주신거 쓰면 더 코드가 짧고 간결해질것 같습니다 ㅎㅎㅎㅎ

k-cool commented on src/pages/hwiminKim/Login/LoginForm.js:
```diff
@@ -0,0 +1,44 @@
+  handleId = event => {
+    const isIdValid = event.target.value.includes('@') ? true : false;
+    this.setState({ isIdValid });
+  };
```
깔끔하니 좋군요... 저도 이렇게 할걸그랬어요 ㅠ 아 쓰다보니 발견!
const isIdValid = event.target.value.includes('@'); 까지만 적어주시면 됩니다! feat.장현님

hwibaski replied:
오.. 그렇네요. 한다고 했는데도 그렇군요 ㅋㅋ 감사합니다

k-cool commented on src/pages/hwiminKim/Login/LoginForm.js:
```diff
@@ -0,0 +1,44 @@
+  };
+
+  handlePw = event => {
+    const isPwValid = event.target.value.length > 4 ? true : false;
```
요놈도 위에처럼 ?부터 날리시면 돼요!ㅎㅎㅎ

k-cool reviewed (commented):
항상 간결하고 깔끔한 코드로 잘읽히는 코드였습니당~~~~;)
휘민님 css 한번 잡아내보고 싶었는데 제 눈엔 안보이네요 ㅠㅠ 휘민님 같은 매의 눈을 갖고 싶다요 ㅠㅠ

wonkooklee commented on src/pages/hwiminKim/component/Review.js:
```diff
@@ -0,0 +1,56 @@
+      <li className='reviewItem'>
+        <div className='contentBox'>
+          <p className='individualReviewBox'>
+            <b>작성자</b> <span>{reviewText}</span>
```
😨 : 휘민님은 장장 1시간에 걸쳐 리뷰했기 때문에 따로 적진 않을게요 :) 멋진 코드 잘봤습니다.

hwibaski replied:
👮🏻‍♂️ : 감사했습니다~

wonkooklee reviewed (commented):
😨 : 휘민님은 장장 1시간에 걸쳐 리뷰했기 때문에 따로 적진 않을게요 :) 멋진 코드 잘봤습니다.

hwibaski committed:
Refactor: Peer review 반영
- Integrated Unnecessary components
- Create Heart Icon component
- Create Breadcrumb component
- Renaming Review component to ReviewThread

hwibaski committed:
refactor: delete console.log(), loginForm function

hwibaski committed:
Refactor: Peer Review Applied
- create _variable.scss
- delete newReview state in ReviewList.js
- rename method name, props name in ReviewList.js and ReviewThread.js

JHyeon0 commented on src/pages/hwiminKim/Detail/Detail.js:
```diff
@@ -1,9 +1,37 @@
-    return <div></div>;
+    const product = this.state.products[4];
+    return (
+      <>
+        <section className='Detail'>
+          <TopsNav />
```
TopsNav는 Detail 바깥에 있어야할 컴포넌트인 것 같습니다.
스타일이 겹칠 가능성도 있지만, 이 코드만 보면 Detail section 내부의 Nav인 것으로 생각됩니다.

JHyeon0 commented on src/pages/hwiminKim/Detail/Detail.js:
```diff
@@ -1,9 +1,37 @@
+      <>
+        <section className='Detail'>
+          <TopsNav />
+          <DetailCategoryHeader productData={product} />
+          <DetailCoffeeInfo productData={product} />
```
props로 필요한 데이터만 넘겨주세요. 넘길 때 props이름도 넘겨지는 데이터 변수 이름과 동일하게 넘겨주세요.

JHyeon0 commented on src/pages/hwiminKim/Login/LoginForm.js:
```diff
@@ -0,0 +1,44 @@
+    isPwValid: false,
+  };
+
+  handleId = event => {
+    const isIdValid = event.target.value.includes('@');
+    this.setState({ isIdValid });
+  };
+
+  handlePw = event => {
+    const isPwValid = event.target.value.length > 4;
+    this.setState({ isPwValid });
+  };
```
[원국님에게 남긴 리뷰] 확인해주세요.
self-refactoring에 계산된 속성명을 확인해주세요.

JHyeon0 commented on src/pages/hwiminKim/Login/LoginForm.js:
```diff
@@ -0,0 +1,44 @@
+          placeholder='비밀번호'
+          onChange={this.handlePw}
+        />
+        <LoginBtn validation={this.state} />
```
active = {계산 다해서 boolean값} 넘겨주는 것이 더 깔끔하겠네요.

JHyeon0 commented on src/pages/hwiminKim/List/CoffeeList.js:
```diff
@@ -0,0 +1,26 @@
+    return (
+      <section className='coffeeList'>
+        <ul className='coffeeListItems'>
+          {productData.map(individualProduct => {
```
그냥 product 하셔도 될 것 같습니다! :)

JHyeon0 commented on src/pages/hwiminKim/component/HeartIcon.js:
```diff
@@ -0,0 +1,31 @@
+import { faHeart } from '@fortawesome/free-solid-svg-icons';
+import './HeartIcon.scss';
+
+class HeartIcon extends Component {
+  state = {
+    isHeartToggleOn: false,
+  };
```
목 데이터로 List.js에서 받아와서 구현해주세요.

JHyeon0 commented on src/pages/hwiminKim/component/Footer.js:
```diff
@@ -0,0 +1,30 @@
+class Footer extends Component {
+  render() {
+    return (
+      <footer className='footer'>
+        <FooterItem category={company} />
+        <FooterItem category={corporateSales} />
+        <FooterItem category={partnership} />
+        <FooterItem category={onlineCommunity} />
+        <FooterItem category={recruit} />
+        <FooterItem category={webucks} />
+      </footer>
```
반복되고 있으니 map 사용해주세요 ~

JHyeon0 commented on src/pages/hwiminKim/Detail/DetailCoffeeInfo.js:
```diff
@@ -0,0 +1,81 @@
+          <p className='nutiritionServingSize'>{nutrition?.servingSize}</p>
+        </div>
+
+        <div className='nutritionContainer'>
+          <div className='nutritionLeft'>
+            <div className='nutritionItembox'>
+              <p>1회 제공량</p>
+              <p>{nutrition?.kcal}</p>
+            </div>
+            <div className='nutritionItembox'>
+              <p>포화지방 (g)</p>
```
table 사용해주세요

JHyeon0 commented on src/pages/hwiminKim/component/ReviewList.js:
```diff
@@ -0,0 +1,66 @@
+    reviews: [],
+  };
+
+  onRemove = id => {
```
removeReview나 deleteReview는 어떠신가요!

JHyeon0 commented on src/pages/hwiminKim/component/ReviewList.js:
```diff
@@ -0,0 +1,66 @@
+import './ReviewList.scss';
+
+class ReviewList extends Component {
+  inputRef = React.createRef();
```
createRef를 사용하지 않아도 구현할 수 있습니다. :)

JHyeon0 commented on src/pages/hwiminKim/component/ReviewList.js:
```diff
@@ -0,0 +1,66 @@
+      this.setState({
+        reviews: reviews.concat({
+          userReview: newReview,
+          id: this.id++,
```
form 태그의 onSubmit 이벤트를 찾아보시면 좋을 것 같아요!

JHyeon0 reviewed (commented):
고생하셨습니다 휘민님!

리뷰 반영해주시고 제가 다른 분들에게 남긴 코드 리뷰도 같이 확인해주시면서 리팩토링 진행해주세요! 화이팅 :)

hwibaski committed:
Refactor: 리뷰 적용

hwibaski committed:
Add: 오탈자 수정
